### PR TITLE
New Exclusions

### PR DESCRIPTION
**Pull request by Lunatixz**
With upcoming Netflix and Amazon plugins there is a user demand to
disable scrobbling when viewing a plugin source... Also there is a need
from script developers to "Pause" trakt scrobbling when needed. After a
user allows script control "script.trakt.paused" property can temp
disable trakt till the media stops or ends then trakt will clear the
property.
 master

### CODE REVIEW

**Comment by razzeee** [comment]
So other scripts will set 'script.trakt.paused' and then we don't start scrobbling. This will only work if that's set before a movie/show starts, correct?

**Comment by Lunatixz** [comment]
A script would place .....`setproperty('script.trakt.paused','true') `before starting playback or immediately after. For example i'd recommend a script use kodi's `player()` class to monitor 'onplaybackstarted` and set the property accordingly.

I placed the job of resetting the property and resuming the scrobbler in trakt's hands to enforce scrobbling.

BTW IC your point and the ONLY issue I see is if a user left your default `scrobble_start_offset` at `0`. This could easily be fixed by settings the minimum to `1` allowing the scripts  'onplaybackstarted` time to set the property.

**Comment by razzeee** [comment]
Sounds good, but I'm not sure if there are any promisses if every subscriber get's the event in the same second. You probably haven't forgotten it, but you are aware that you need to set it again `onplaybackresumed`, correct?
That has no offset, so that might also complicate things.

**Comment by Lunatixz** [comment]
True, I like to keep things simple... and this was the simplest solution I could think of. We/I can make this as involved as you want? The important thing is to allow scripts a temporary opt out... 

From what I remember trakts `onplaybackresumed `relied on `self._playing` to be true... however its set at false after `checkExclusion()`?

**Comment by Lunatixz** [comment]
Should I look for a alternative approach? or do you think this is viable?

**Comment by razzeee** [comment]
Well have you runtime tested it? Does it work like you want it to work? Does it resume like expected? The code looks not harmful :)

**Comment by Lunatixz** [comment]
Yes i've tested it with a script for a few weeks, works as expected. As long as there is a `1` sec `scrobble_start_offset `. If its set to `0` its anyones guess who's `onplaybackstarted` service will execute first.

**Comment by razzeee** [comment]
Let's hardcode a offset in the codebase then?

**Comment by Lunatixz** [comment]
I'm not sure it's needed on further review of the code https://github.com/trakt/script.trakt/blob/master/resources/lib/service.py#L373

It looks like there is already a 1sec sleep, if you want to lean on the safer side we could bump that to 2000 msec but I believe that's overkill.

File: resources/language/English (US)/strings.po [hunk 1]
```diff
@@ -643,3 +643,11 @@ msgstr "Trakt - Added to watchlist"
 msgctxt "#32166"
 msgid "Trakt - Failed adding to watchlist"
 msgstr "Trakt - Failed adding to watchlist"
+
```
Comment (razzeee) [inline resources/language/English (US)/strings.po]: Don't change this file, it will get overwritten if we get any language updates. (someday)

Comment (Lunatixz) [reply resources/language/English (US)/strings.po]: Only this particular language po, or all language strings? What would you recommend?

Comment (razzeee) [reply resources/language/English (US)/strings.po]: This one the English gb has to be there. Everything else is based upon it.

**Comment by razzeee** [comment]
So one last thing travis / the tests are not happy anymore.
As your adding xbmcgui it needs to mock them in the test class

https://github.com/trakt/script.trakt/blob/master/tests/test_kodiUtilities.py#L11 for reference

**Comment by Lunatixz** [comment]
THX, I can't locate mock or pytest...

**Comment by razzeee** [comment]
Yes, you will have to install them locally via pip. See for reference:
https://github.com/trakt/script.trakt/blob/master/.travis.yml

**Comment by Lunatixz** [comment]
@Razzeee i'm not familiar with mock and its syntax... if you could assist me I would appreciate it. Looks like my change created a conflict.

**Comment by razzeee** [comment]
Okay, just roll back your change I will merge as it is and fix the tests later :)

**Comment by razzeee** [comment]
Can you rebase on master? Github doesn't seem to be able to do so

**Comment by Lunatixz** [comment]
want me to close the commit and reissue with a clean base?

**Comment by razzeee** [comment]
Sure, if your not able to get this branch working :)
Just rebase (maybe squash) and force push should do the trick

**Comment by Lunatixz** [comment]
I rebased and forced push, everything is up to date... not sure what the issue is.... Going to start clean

**Pull request closed without merging**